duckduckgo/macos-browser
# Implements new onboarding UI/UX

Task/Issue URL: https://app.asana.com/0/0/1205257644127197/f

## Stack

- Feature branch: `diego/feature-network-protection-onboarding`
    - https://github.com/duckduckgo/macos-browser/pull/1474
        - https://github.com/duckduckgo/macos-browser/pull/1464 (this PR)

## Description

Implements the new onboarding UI/UX.

## Scope

The UI is designed here: https://www.figma.com/file/nscneJs8hvI6j2m24AFTsD/Setup-steps-macOS-Network-Protection?type=design&node-id=132%3A9918&mode=design&t=qrWZTHgdsccYM1IZ-1

My suggestion though would be not to focus on the UI too strongly unless you see a big issue, as this will be run through a design review anyway.

### In Scope:

 - Light mode
 - Showing the UI that signals the user needs to allow the system extension, before the system extension is allowed
 - Showing the UI that signals the user needs to allow the VPN configuration, before the VPN configuration is allowed 

### Out of Scope:

- Dark mode
- Different images for older macOS versions

## Steps to test this PR:

1. Launch the app
2. Reset Network Protection's status to make sure you start with a clean state.
3. Kill the menu app through Activity Monitor at least once
4. Open the NetP popover and make sure it shows a view that shows how to allow the system extension.  The controller part of the view should be disabled.
5. Click on the button in that view, NetP should start and request that you allow the system extension
6. Allow it
7. Now the UI (in the popover and menu app) should be updated to show a view that explains how to allow the VPN configuration.  The controller part of the view should be disabled.
8. Allow it
9. The onboarding UI should go away and you should see the regular NetP popover status view. 

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
